File: solutions/002/2.rb
# Находим сумму чётных элементов ряда Фиббоначи до 4млн (4_613_732)

old, new, rez = 1, 2, 0
while new <= 4_000_000
  rez += new if new.even?
  old, new = new, old + new
end
p rez

File: solutions/002/2.hs
-- 4613732

fibs = map fst $ iterate (\(x, y) -> (y, x + y)) (0, 1)

sumEven :: [Int] -> Int
sumEven = foldr (\x sum -> if even x then x + sum else sum) 0

main :: IO ()
main = do
    print $ sumEven $ takeWhile (< 4000000) fibs
